superset/db_engine_specs/base.py

```diff
@@ -649,6 +653,57 @@ def apply_limit_to_sql(
 
         return sql
 
+    @classmethod
+    def apply_top_to_sql(
+        cls, sql: str, _limit: int, database: str = "Database", force: bool = False
+    ) -> str:
+        """
+        Alters the SQL statement to apply a TOP clause
+        :param _limit: Maximum number of rows to be returned by the query
+        :param sql: SQL query
+        :param database: Database instance
+        :return: SQL query with top clause
+        """
+        sql_statement = sqlparse.format(sql, strip_comments=True)
+        query_limit: Optional[int] = sql_parse._extract_top_from_query(sql_statement)
+        td_sel_keywords = {"SELECT","SEL"}
+        td_top_keywords = {"TOP","SAMPLE"}
```

Comment:
`SEL` and `SAMPLE` seem to be specific to Teradata. We should probably move these into `BaseEngineSpec` like this:

```python
class BaseEngineSpec:
    ...
    # all keywords that should be treated as SELECTs
    select_keywords = {"SELECT"}
    # For engines that use TOP instead of LIMIT keyword, 
    list all keywords that should be treated similar top TOP keywords
    top_keywords = {"TOP"}
```

Then for Teradata these could be set to
```python
    select_keywords = {"SELECT","SEL"}
    top_keywords = {"TOP","SAMPLE"}
```

and for MSSQL it would probably be ok to go with the defaults (I assume `SEL` and `SAMPLE` aren't valid for MSSQL).

Reply:
Made this change